# HG changeset patch
# User John Donoghue
# Date 1428330560 14400
#      Mon Apr 06 10:29:20 2015 -0400
# Node ID c66909c234e6771c6c7d185b11c6642a2f5df117
# Parent  91e1da1d19183c23cfae2807c2448e131a665ec1
Allow bool_scalar types for graphics array properties (Bug #44699)

* libinterp/corefcn/graphics.cc
  (array_property::validate): set xok to true also if is bool_scalar

diff --git a/libinterp/corefcn/graphics.cc b/libinterp/corefcn/graphics.cc
--- a/libinterp/corefcn/graphics.cc
+++ b/libinterp/corefcn/graphics.cc
@@ -1328,17 +1328,17 @@ array_property::validate (const octave_v
 
       // check if complex is allowed (it's also of class "double", so
       // checking that alone is not enough to ensure real type)
       if (type_constraints.find ("real") != type_constraints.end ()
           && v.is_complex_type ())
         xok = false;
     }
   else
-    xok = v.is_numeric_type ();
+    xok = v.is_numeric_type () || v.is_bool_scalar ();
 
   if (xok)
     {
       if (size_constraints.size () == 0)
         return true;
 
       dim_vector vdims = v.dims ();
       int vlen = vdims.length ();
